File: src/main/java/io/github/takenoha/towerdefense/paper/CoreRecipeDefinition.java
package io.github.takenoha.towerdefense.paper;

import java.util.List;

/** Vanilla ingredients for the public core item recipe. */
public final class CoreRecipeDefinition {
    public static final String DIAMOND_BLOCK_MATERIAL = "DIAMOND_BLOCK";
    public static final String IRON_INGOT_MATERIAL = "IRON_INGOT";

    private CoreRecipeDefinition() {
    }

    public static List<String> shape() {
        return List.of(" I ", "IDI", " I ");
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/paper/CoreRecipeDefinition.kt
package io.github.takenoha.towerdefense.paper

/** Vanilla ingredients for the public core item recipe. */
class CoreRecipeDefinition private constructor() {
    companion object {
        const val DIAMOND_BLOCK_MATERIAL: String = "DIAMOND_BLOCK"
        const val IRON_INGOT_MATERIAL: String = "IRON_INGOT"

        @JvmStatic
        fun shape(): List<String> = listOf(" I ", "IDI", " I ")
    }
}
